chapters/chap03-sec03.tex

```diff
@@ -36,25 +36,25 @@ \section{\zeth~statement after primitive instantiation}\label{instantiation:stat
-Indeed, the cost of $\groth$ zk-SNARK~\cite{groth2016size} proof verification is linear in the number of primary inputs since each input acts as a scalar in a costly scalar multiplication of an element in $\gset_1$. Hence, while packing puts ``(slightly) more pressure'' on the prover -- by adding constraints in the circuit -- it simplifies the verifier's work.
+As we consider SNARKs based on arithmetic circuits defined over a prime field, all variables in the constraint system are interpreted as field elements. Nevertheless, as illustrated in~\cref{zeth-protocol:statement}, part of the statement consists of functions whose co-domains are sets of binary strings (which may be longer than the bit representation of elements of the finite field). While a bit (i.e.~$\{0, 1\}$) is an element of $\FF_p$ ($p$ prime), it is important to minimize the number of gates in the arithmetic circuit (for proof generation efficiency), and to minimize the number of input wires (to improve verification time). This can be done by representing fragments of binary strings as the base 2 decomposition of binary elements, thereby ``packing'' binary strings into multiple elements. Converting binary strings into field elements requires the addition of some arithmetic gates (extending the statement to be proven), but reduces the number of primary inputs (reducing the complexity of the SNARK verification carried out on-chain).
+The cost of $\groth$ zk-SNARK~\cite{groth2016size} proof verification is linear in the number of primary inputs, since each input acts as a scalar in a costly scalar multiplication of a curve point in $\gset_1$. Hence, while packing slightly increases the prover cost -- by adding constraints to the circuit -- it simplifies the verifier's work.
 
-We detail in this section the method by which we encode (resp.~decode) a set of binary strings (resp.~field elements) into a set of field elements (resp.~binary strings). In the rest of this section, the notion of \emph{packing policy} is used to refer to the set of \emph{packing} and \emph{unpacking} functions.
+In this section we detail the method by which we encode (resp.~decode) a set of binary strings to (resp.~from) sets of field elements. In the rest of this section, the notion of \emph{packing policy} refers to the set of \emph{packing} and \emph{unpacking} functions.
```

Comment:
`In this section` -> `In this section,`?

Reply:
Changed.